### PR TITLE
fix(ct): pass local config to preview

### LINKED ISSUE #16460: [BUG] Component test fails on 2nd run when SSL is used
**Context:**
- Playwright Version: `"playwright/experimental-ct-svelte": "^1.24.2"`
- Operating System: macOS
- Node.js version: 16.14.2
- Browser: All
- Extra: ARM chip device

**Code Snippet**

Here's the reduced test case which reproduces the issue:
https://github.com/vhscom/playwright-ct-ssl-reduced-test

I've left the reduced test case configured with the Vite plugin for SSL but the issue can be reproduced without plugins specified in the Playwright component config file.

**Describe the bug**

This is an extension of https://github.com/microsoft/playwright/issues/16424. When testing a Svelte component using `@vitejs/plugin-basic-ssl` plugin test runner fails with error message when cache is present and unless the component under test has a timestamp modification (read: it was saved, even without changes):

```
undefined: net::ERR_EMPTY_RESPONSE at http://localhost:3100/playwright/index.html
=========================== logs ===========================
navigating to "http://localhost:3100/playwright/index.html", waiting until "load"
============================================================
```

I narrowed it down to a TLS/vite config/playwright lib corner case. It seems the URL under test does not work when the `@vitejs/plugin-basic-ssl` plugin is used—but only on the second run if the component under test doesn't have a timestamp modification since the last cached run.

Ideally tests will either fail wholesale if SSL is configured or there will be a way to run component tests over the `https` scheme—which would be expected when the plug-in is configured via Playwright experimental `ctViteConfig` prop. Having test runs once, fail on the second run, then work again when the cache is cleared or the component timestamp changed is unexpected behavior.

### PR DESCRIPTION
**Pull request by pavelfeldman**
Fixes https://github.com/microsoft/playwright/issues/16460

### CODE REVIEW

**Review by rwoll — APPROVED** [review]
Bump your

```
const VERSION = 6;
```

to `7` for good measure?

**Comment by rwoll** [comment]
Shall we also add a test?

**Review by dgozman — APPROVED** [review]

**Review by dgozman — APPROVED** [review]

File: packages/playwright-core/src/cli/driver.ts [hunk 1]
```diff
@@ -197,10 +197,16 @@ class ProtocolHandler {
...
+  id: number;
+  method: string;
+  params: Record<string, any>
+};
+
 function wireController(playwright: Playwright, wsEndpoint: string) {
   process.send!({ method: 'ready', params: { wsEndpoint } });
   const handler = new ProtocolHandler(playwright);
-  process.on('message', async message => {
+  process.on('message', async (message: DriverProtocolMessage) => {
```
Comment (rwoll) [inline packages/playwright-core/src/cli/driver.ts:209]: Is this and the @types/node change needed in this PR?

**Review by rwoll — APPROVED** [review]
LGTM! Thanks for adding the test!

File: packages/playwright-core/src/cli/driver.ts (hunk 1 above)
Comment (rwoll) [reply packages/playwright-core/src/cli/driver.ts:209]: Ignore—we chatted offline
